packages/remix-dev/cli/convert-to-javascript.ts

```diff
@@ -0,0 +1,91 @@
+import path from "path";
+import fse from "fs-extra";
+import glob from "fast-glob";
+import * as babel from "@babel/core";
+// @ts-expect-error these modules dont have types
+import babelPluginSyntaxJSX from "@babel/plugin-syntax-jsx";
+// @ts-expect-error these modules dont have types
+import babelPresetTypeScript from "@babel/preset-typescript";
+import prettier from "prettier";
+
+function convertToJavaScript(
+  filename: string,
+  source: string,
+  projectDir: string
+): string {
+  let result = babel.transformSync(source, {
+    filename,
+    presets: [[babelPresetTypeScript, { jsx: "preserve" }]],
+    plugins: [babelPluginSyntaxJSX],
+    compact: false,
+    retainLines: true,
+    cwd: projectDir,
+  });
+
+  if (!result || !result.code) {
+    throw new Error("Could not parse typescript");
+  }
+
+  /*
+    Babel's `compact` and `retainLines` options are both bad at formatting code.
+    Use Prettier for nicer formatting.
+  */
+  return prettier.format(result.code, { parser: "babel" });
+}
+
+export async function convertTemplateToJavaScript(projectDir: string) {
+  // 1. Convert all .ts files in the template to .js
+  let entries = glob.sync("**/*.+(ts|tsx)", {
+    cwd: projectDir,
+    absolute: true,
+    ignore: ["**/node_modules/**"],
+  });
+  for (let entry of entries) {
+    if (entry.endsWith(".d.ts")) {
+      fse.removeSync(entry);
+      continue;
+    }
+
+    let contents = fse.readFileSync(entry, "utf8");
+    let filename = path.basename(entry);
+    let javascript = convertToJavaScript(filename, contents, projectDir);
+
+    fse.writeFileSync(entry, javascript, "utf8");
+    if (entry.endsWith(".tsx")) {
+      fse.renameSync(entry, entry.replace(/\.tsx?$/, ".jsx"));
+    } else {
+      fse.renameSync(entry, entry.replace(/\.ts?$/, ".js"));
+    }
+  }
+
+  // 2. Rename the tsconfig.json to jsconfig.json
+  if (fse.existsSync(path.join(projectDir, "tsconfig.json"))) {
+    fse.renameSync(
+      path.join(projectDir, "tsconfig.json"),
+      path.join(projectDir, "jsconfig.json")
+    );
+  }
+
+  // 3. Remove @types/* and typescript from package.json
+  let packageJson = path.join(projectDir, "package.json");
+  if (!fse.existsSync(packageJson)) {
+    throw new Error("Could not find package.json");
+  }
+  let pkg = JSON.parse(fse.readFileSync(packageJson, "utf8"));
+  let devDeps = pkg.devDependencies || {};
+  let newPackageJson = {
+    ...pkg,
+    devDependencies: Object.fromEntries(
+      Object.entries(devDeps).filter(([name]) => {
+        return !name.startsWith("@types/") && name !== "typescript";
+      })
+    ),
+  };
+  // 4. Remove typecheck npm script from package.json
+  if (pkg.scripts && pkg.scripts.typecheck) {
+    delete pkg.scripts.typecheck;
+  }
```

Comment:
This doesn't remove the `typecheck` script from `newPackageJson`, so does the removal of the `typecheck` script ever get written to `package.json`?

Reply:
it does now after [dde9799](https://github.com/remix-run/remix/pull/2556/commits/dde9799390d844a3488a683b5cb50e92a785645e), but it seems to have still removed it using the 1.3.4 on the blues stack with `--no-typescript` - but looks like we should maybe also check each script to see if one of them also calls "typecheck" like the blues stack does `"validate": "run-p \"test -- --run\" lint typecheck test:e2e:run"`

Reply:
I think we can let the remix.init script handle that. It's impossible to handle every case for stacks authors, so we just do our best here and then stacks authors can do the rest